zips/draft-stark-verify-nu.md

```diff
@@ -0,0 +1,150 @@
+    ZIP: XXX
+    Title: Circle STARK Verification as a Transparent Zcash Extension
+    Owners: Abdel <abdel.dev.bitcoin@proton.me>, Michael Zaikin <michael.z@starkware.co>
+    Status: Draft
+    Category: Consensus
+    Created: 2025-10-14
+    License: MIT
+    Discussions-To: <https://forum.zcashcommunity.com/t/stark-verification-as-a-transparent-zcash-extension-tze-i-e-enabler-for-a-starknet-style-l2-on-zcash/52486>
+
+# Terminology
+
+The key words "MUST", "SHOULD", and "MAY" are to be interpreted as described in BCP 14 [^BCP14] when, and only when, they appear in all capitals.
+
+## Abstract
+
+This ZIP defines a new Transparent Zcash Extension that verifies a bounded‑size Circle STARK proof (Stwo). The extension has a single `type` and a single mode, and exposes a compact precondition/witness interface suitable for verifying L2 validity proofs on Zcash L1. It specifies:
```

Comment:
"L1" and "L2" are not defined (and I don't want readers to rely on intuition from other chains because I think it will be useful to define them using other Zcash terminology).